src/main/kotlin/com/github/iwanabethatguy/oxcintellijplugin/lsp/OxcLspServerDescriptor.kt

```diff
@@ -1,37 +1,59 @@
 package com.github.iwanabethatguy.oxcintellijplugin.lsp
 
-import com.intellij.execution.ExecutionException
 import com.intellij.execution.configurations.GeneralCommandLine
 import com.intellij.openapi.diagnostic.thisLogger
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.vfs.VirtualFile
 import com.intellij.platform.lsp.api.ProjectWideLspServerDescriptor
-import com.intellij.openapi.fileTypes.FileType
-import com.intellij.platform.lsp.api.customization.LspDiagnosticsSupport
-import com.intellij.platform.lsp.api.requests.LspRequest
-import org.eclipse.lsp4j.ConfigurationItem
-import org.eclipse.lsp4j.InitializeParams
+import com.intellij.util.FileSearchUtil
+import java.time.Duration
 
+@Suppress("UnstableApiUsage")
 class OxcLspServerDescriptor(project: Project) : ProjectWideLspServerDescriptor(project, "Oxc") {
 
-    override fun isSupportedFile(file: VirtualFile):  Boolean {
-        thisLogger().warn("file.extension " + file.extension.toString())
-        return file.extension == "js" ||  file.extension == "jsx" || file.extension == "ts" || file.extension == "tsx"
+    companion object {
+
+        fun supportedFile(file: VirtualFile): Boolean {
+            return file.isInLocalFileSystem && listOf("js", "jsx", "ts", "tsx").contains(
+                file.extension)
+        }
+    }
+
+    override fun isSupportedFile(file: VirtualFile): Boolean {
+        thisLogger().warn(
+            "file.isInLocalFileSystem ${file.isInLocalFileSystem}, file.extension ${file.extension}")
+        return supportedFile(file)
     }
 
     override fun createCommandLine(): GeneralCommandLine {
         thisLogger().warn("Start oxc language server")
 
-        return GeneralCommandLine("oxc_language_server").apply {
-        }
+        return GeneralCommandLine(findBinary())
     }
 
     override fun createInitializationOptions(): Any? {
         val options = super.createInitializationOptions()
-        thisLogger().warn("get oxc configuration" + options.toString())
+        thisLogger().warn("get oxc configuration $options")
         return options
     }
 
+    private fun findBinary(): String {
+        val foundBinaries = roots.map {
+            val fileQuery = FileSearchUtil.findFileRecursively(it, "oxc_language_server", 10,
```

Comment:
10 is the depth to search
5 seconds is the timeout

Both of these values were pretty arbitrarily picked and can be adjusted as needed.  Maybe a possible user setting in the future.